src/WorkItemMigrator/JiraExport/JiraProvider.cs

```diff
@@ -405,25 +395,40 @@ public string GetUserEmail(string usernameOrAccountId)
         public string GetCustomId(string propertyName)
         {
             var customId = string.Empty;
-            var response = (JArray)Jira.RestClient.ExecuteRequestAsync(Method.GET, $"{JiraApiV2}/field").Result;
-            foreach (var item in response)
+            if (JiraNameFieldCache == null)
             {
-                var nameField = (JValue)item.SelectToken("name");
-                if (nameField.Value.ToString().ToLower() == propertyName.ToLower())
-                {
-                    var idField = (JValue)item.SelectToken("id");
-                    customId = idField.Value.ToString();
-                }
+                var response = (JArray)_jiraServiceWrapper.RestClient.ExecuteRequestAsync(Method.GET, $"{JiraApiV2}/field").Result;
+                JiraNameFieldCache = response
```

Comment:
I dont follow the duplication here ?  from line 400 to 410. 
Why do you need the key as well ?

Reply:
My Jira instance does not return a "key" property on any of the fields. However, the existing code checks the "name" property first and if no match is found it checks the "key" property. Does that answer your question?

Reply:
@moerketh

Reply:
I mean, I would move the KeyCache inside the if statement where we actually try to use it and there check if the key cache is null or not and then fill it.